pkg/loki/config_wrapper.go

```diff
@@ -99,12 +99,12 @@ func (c *ConfigWrapper) ApplyDynamicConfig() cfg.Source {
 			return err
 		}
 
-		if len(r.SchemaConfig.Configs) > 0 && config.UsingBoltdbShipper(r.SchemaConfig.Configs) {
-			betterBoltdbShipperDefaults(r, &defaults)
+		if i := lastBoltdbShipperConfig(r.SchemaConfig.Configs); i != len(r.SchemaConfig.Configs) {
```

Comment:
What was the reason for using `len(r.SchemaConfig.Configs` as your sentinel value when a config can't be found here? I wonder if `-1` might be a more canonical?

Reply:
I used the convention from [`sort.Search`](https://go.dev/src/sort/search.go) in the stdlib.

Reply:
ahh, all these go conventions I don't know. sounds good!